### PR TITLE
Issue/672 verbosecli

### PR DESCRIPTION
**Pull request by sagar2395**
#672 Added comments in CLI to make it more verbose

### CODE REVIEW

**Review by ChrisSchinnerl — CHANGES_REQUESTED** [review]

File: cmd/renterd/main.go [hunk 1]
```diff
@@ -138,7 +140,8 @@ func getSeed() types.PrivateKey {
 	if seed == nil {
 		phrase := cfg.Seed
 		if phrase == "" {
-			fmt.Print("Enter seed: ")
+			fmt.Println("Type in your 12-word seed phrase and press enter. If you do not have a seed phrase yet, type 'seed' to generate one")
```
Comment (peterjan) [inline cmd/renterd/main.go:143]: `nit` I would inform the user he can generate a seed using the `renterd seed` command if he doesn't have one, but printing it like this I think some people would type seed here, and then be hit with `wrong number of words in seed phrase`

File: cmd/renterd/main.go [hunk 2]
```diff
@@ -321,8 +341,24 @@ func main() {
 		log.Println("Build Date:", builddate)
 		return
 	} else if flag.Arg(0) == "seed" {
-		log.Println("Seed phrase:")
-		fmt.Println(wallet.NewSeedPhrase())
+		var seed [32]byte
```
[the remark this answers is not in the document]
Comment (peterjan) [reply cmd/renterd/main.go:344]: I was actually going to suggest doing this by default. I think having a verbose CLI is fine but a tool that generates a seed should print 12 words and that's it. Currently we print something like:

```
renterd v0.6.0
Network Mainnet
Seed phrase:
```

I think printing the address is a nice addition though, but maybe we can print that every time `renterd` is started?
What if instead of a `--headless` flag we add a `--help` that gives some more ELI5 info about the seed and what it's used for and why you should make sure you have it backed up.

File: cmd/renterd/main.go [hunk 3]
```diff
@@ -125,7 +125,9 @@ func mustLoadAPIPassword() {
 		return
 	}
 
-	fmt.Print("Enter API password: ")
+	fmt.Println("Please choose a password to unlock the UI and API.")
+	fmt.Println("(The password must be at least 4 characters.)")
```
Comment (peterjan) [inline cmd/renterd/main.go:129]: If we're not asserting these password criteria we shouldn't be printing them I think. I would personally find it a little annoying to have password requirements but at the same time it's probably a good idea, in which case we should use more meaningful criteria.

File: wallet/seed.go [hunk 4]
```diff
@@ -48,6 +48,28 @@ func KeyFromPhrase(phrase string) (types.PrivateKey, error) {
 	return key, nil
 }
 
+func SeedFromPhrase(seed *[32]byte, phrase string) error {
```
Comment (peterjan) [inline wallet/seed.go:51]: reminder: remove this after switching to `KeyFromPhrase`

File: cmd/renterd/main.go (hunk 1 above)
Comment (n8maninger) [reply cmd/renterd/main.go:143]: They should be able to type “seed” and generate on if this is similar to `hostd`

Comment (peterjan) [reply cmd/renterd/main.go:143]: Oh, well that's fine too but then we should check for it because right now I think you'd be hit with `wrong number of words in seed phrase`. If that is not the case please feel free to resolve my comment, I haven't built it yet to try it out myself, but I will once this round of comments is done.

File: cmd/renterd/main.go [hunk 5]
```diff
@@ -321,8 +341,24 @@ func main() {
...
+		newPhrase := wallet.NewSeedPhrase()
+		fmt.Println("Seed phrase:", newPhrase)
+
+		if err := wallet.SeedFromPhrase(&seed, newPhrase); err != nil {
+			panic(err)
+		}
+
+		privKey := wallet.KeyFromSeed(&seed, 0)
+		address := privKey.PublicKey().StandardAddress()
+		fmt.Println("Wallet", address)
```
[the remark this answers is not in the document]
Comment (sagar2395) [reply cmd/renterd/main.go:359]: the address variable is printing "addr:<wallet address>". This is why I just added wallet in front of that, which makes it 
"wallet addr: <address>". Should I slice the addr part and update it to wallet funding address ?

File: cmd/renterd/main.go [hunk 6]
```diff
@@ -321,8 +341,24 @@ func main() {
...
+		fmt.Println("You will need to re-enter this seed phrase every time you start renterd.")
+
+		newPhrase := wallet.NewSeedPhrase()
+		fmt.Println("Seed phrase:", newPhrase)
+
+		if err := wallet.SeedFromPhrase(&seed, newPhrase); err != nil {
+			panic(err)
+		}
+
+		privKey := wallet.KeyFromSeed(&seed, 0)
```
[the remark this answers is not in the document]
Comment (sagar2395) [reply cmd/renterd/main.go]: Done!!

File: wallet/seed.go (hunk 4 above)
Comment (sagar2395) [reply wallet/seed.go]: Done!!

File: wallet/seed.go [hunk 7]
```diff
@@ -48,6 +48,28 @@ func KeyFromPhrase(phrase string) (types.PrivateKey, error) {
...
+		return err
+	}
+	h := blake2b.Sum256(entropy[:])
+	memclr(entropy[:])
+	copy(seed[:], h[:])
+	memclr(h[:])
+	return nil
+}
+
+func KeyFromSeed(seed *[32]byte, index uint64) types.PrivateKey {
```
[the remark this answers is not in the document]
Comment (sagar2395) [reply wallet/seed.go]: Done!!

File: cmd/renterd/main.go (hunk 3 above)
Comment (sagar2395) [reply cmd/renterd/main.go:129]: @peterjan Can you please confirm if we should keep this criteria or can you provide the criteria details that should be updated here ?

File: cmd/renterd/main.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (sagar2395) [reply cmd/renterd/main.go]: Slice part is added. Kindly review!!

File: cmd/renterd/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sagar2395) [reply cmd/renterd/main.go]: Changing nothing for this. Kindly resolve this or let me know if any action is required

File: cmd/renterd/main.go (hunk 3 above)
Comment (peterjan) [reply cmd/renterd/main.go:129]: I would drop it for now. I think it's fine to introduce password criteria but perhaps that can be a separate PR. At the very least if we print the requirement we should assert the length of the password somewhere.

**Comment by sagar2395** [comment]
@peterjan @ChrisSchinnerl @n8maninger I have made the requested changes. Kindly review and let me know if any other change is required

File: cmd/renterd/main.go (hunk 1 above)
Comment (peterjan) [reply cmd/renterd/main.go:143]: You have to check whether the user's input equals "seed" because we've just instructed the user to type "seed" if he wants to generate one. You can look at [hostd](https://github.com/SiaFoundation/hostd/blob/master/cmd/hostd/main.go#L104) for inspiration.

File: cmd/renterd/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (peterjan) [reply cmd/renterd/main.go]: I think we want to make the behaviour consistent with `hostd`, meaning that if the user types "seed" we'll generate one for him. Chris is suggesting that _if we do that_ (which we do) it's fine to only print the seed here, which we are currently not doing. This is important because these tools are often used for scripting and we don't want everyone to have to parse out the seed from the command's output, but instead the command should only print the seed and nothing more.

**Review by peterjan — CHANGES_REQUESTED** [review]
On the right track, just some minor comments left. I left some links to `hostd` for inspiration.

**Comment by peterjan** [comment]
@ChrisSchinnerl I vote we close this out - we can always reopen if necessary

**Comment by ggwpez** [comment]
Some (possibly) unwanted user-feedback: the experience for new users is still quite poor. I though i may need to register for an API key or such just to run the software when reading then print.
